client/core/core_test.go

```diff
@@ -352,7 +353,32 @@ func (tdb *TDB) LinkOrder(oid, linkedID order.OrderID) error {
 	return nil
 }
 
+func (tdb *TDB) setUpdateMatchHook(ch chan order.MatchStatus) {
+	tdb.updateMatchMtx.Lock()
+	defer tdb.updateMatchMtx.Unlock()
+	tdb.updateMatchChan = ch
+}
+
+func (tdb *TDB) waitForMatchUpdate() (order.MatchStatus, error) {
+	tdb.updateMatchMtx.RLock()
+	defer tdb.updateMatchMtx.RUnlock()
+	var latestStatus order.MatchStatus
+	if tdb.updateMatchChan == nil {
+		return latestStatus, errors.New("not configured to track match db updates")
+	}
+	select {
+	case latestStatus = <-tdb.updateMatchChan:
+		return latestStatus, nil
+	case <-time.After(1 * time.Second):
+		return latestStatus, errors.New("no match db update in 1 second")
+	}
+}
```

Comment:
The previous concurrent reading and writing of `tdb.updateMatchChan` caused some panics for me. Introduced the mutex lock and these helper functions to prevent littering the file with mutex locks in each site where `tdb.updateMatchChan` is set or accessed.